src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketRepositoryClientImpl.java

```diff
@@ -50,6 +66,30 @@ public BitbucketBuildStatusClient getBuildStatusClient(String revisionSha,
         return new BitbucketBuildStatusClientImpl(bitbucketRequestExecutor, revisionSha);
     }
 
+    @Override
+    @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
+    public BitbucketBranchClient getBranchClient(TaskListener listener, SCMSourceOwner context, BitbucketSCMRepository repository) throws IOException, InterruptedException {
```

Comment:
I think it might be worth considering a new package for git clients or as part of our `scm` package, rather than mixing them with our HTTP clients. All git requests will need to be made against a single repository so I presume the client all clients will look pretty similar if provided these two things, so it might just be a matter of adding methods for the different commands we're calling, at least for now.